File: tests/fsharpqa/Source/Import/E_SealedMethod.cs
using System;
namespace ClassLibrary1
{
    public class Class1
    {
        public virtual void F()
        {
            Console.WriteLine("class1");
        }
    }
    public class Class2 : Class1
    {
        public override sealed void F()
        {
            Console.WriteLine("class2");
        }
    }
/*
    public class Class3 : Class2
    {
        public override void F()
        {
            Console.WriteLine("class2");
        }
    }
 */
}

File: tests/fsharpqa/Source/Import/E_SealedMethod.fs
// #Regression #NoMT #Import 
// Dev11 Bug 90642
//<Expects status="error" span="(8,19)" id="FS3070">Cannot override inherited member 'ClassLibrary1.Class2::F' because it is sealed$</Expects>

type MyClass() =
    inherit ClassLibrary1.Class2()
 
    override this.F() = 
        printfn "aha!"
 
let x = new MyClass() 
let y = x :> ClassLibrary1.Class1 
y.F()
